RMtools_3D/do_fitIcube.py

```diff
@@ -70,91 +77,166 @@ def main():
                                  formatter_class=argparse.RawTextHelpFormatter)
     parser.add_argument("fitsI", metavar="StokesI.fits", nargs=1,
                         help="FITS cube containing Stokes I data.")
-    parser.add_argument("freqFile", metavar="freqs_Hz.dat", nargs=1,
-                        help="ASCII file containing the frequency vector.")
+    parser.add_argument("-freq", dest="freq_file", default="",type=str,
+                       help="Path + ASCII file containing the frequency vector. If not provided,\nfrequencies are derived from fits header.")
     parser.add_argument("-f", dest="fit_function", type=str, default="log",
                         help="Stokes I fitting function: 'linear' or ['log'] polynomials.")
-    parser.add_argument("-p", dest="polyOrd", type=int, default=2,
+    parser.add_argument("-or", dest="polyOrd", type=int, default=2,
                         help="polynomial order to fit to I spectrum: 0-5 supported, 2 is default.\nSet to negative number to enable dynamic order selection.")
-    parser.add_argument("-c", dest="cutoff", type=float, default=-3,
-                        help="emission cutoff (+ve = abs, -ve = sigma) [-3].")
-    parser.add_argument("-o", dest="prefixOut", default="",
-                        help="Prefix to prepend to output files [None].")
+    parser.add_argument("-th", dest="threshold", type=float, default=-5,
+                        help="Noise cutoff threshold (+ve = abs, -ve = sigma) [-5].")
+    parser.add_argument("-nc", dest="num_cores", type=int, default=10,
+                        help="Number of cores to use for multiprocessing. Default is 10.")
+    parser.add_argument("-mk", dest="apply_mask", action='store_true',
+                        help="Apply masking before spectral fitting. Default is False.")                   
+    parser.add_argument("-pf", dest="prefixOut", default="",
+                        help="Prefix to use for to output file names.")
+    parser.add_argument("-od", dest="outDir", default="",
+                        help="Output directory to save output files. If none, save inside input directory")
```

Comment:
While I can appreciate the consistency of making all the inputs 2 characters, I'm not sure it's good to change the existing flags (`-p` to `-or`, for example) -- it will break any scripts that people have set up already that use the old flags. One rule of software engineering is not to change things without a good reason, because it will break things for users. It's fine for the new flags to be 2 characters, but can you change the old flags back to what they were?